src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAYLIST_SIZE = 10;
+    private int size;
+
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[])(new Object[DEFAULT_ARRAYLIST_SIZE]);
+    }
+
     @Override
     public void add(T value) {
-
+        if (size + 1 > array.length) {
+            growArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        outOfRangeCheck((index == size && size != 0) ? (index - 1) : index);
+        if (size + 1 > array.length) {
+            growArray();
+        }
```

Comment:
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
        if (size == array.length) {
            growArray();
        }
```
if index == size you can easily reuse add(T value) method
rename method, because this method checks the index
size == array.length is literally the same as size + 1 > array.length, but less confusing because + 1 makes it less obvious, what is checked in this if